mesonbuild/coredata.py

```diff
@@ -568,15 +568,7 @@ def set_options(self, opts_to_set: T.Dict[OptionKey, T.Any], subproject: str = '
     def add_compiler_options(self, c_options: MutableKeyedOptionDictType, lang: str, for_machine: MachineChoice,
                              env: Environment, subproject: str) -> None:
         for k, o in c_options.items():
-            value = env.options.get(k)
-            if value is not None:
-                o.set_value(value)
-                if not subproject:
-                    # FIXME, add augment
-                    #self.optstore[k] = o  # override compiler option on reconfigure
-                    pass
-
-            comp_key = OptionKey(f'{k.name}', None, for_machine)
+            comp_key = OptionKey(k.name, None, for_machine)
```

Comment:
```suggestion
            comp_key = comp_key = k.evolve(subproject=None, machine=for_machine)
```

Reply:
I hope we could even assert that `k` already has the right shape. If it doesn't work, then for 1.8.0 I agree `evolve` is better (and it can then be fixed later).

Reply:
I removed the change altogether because of failures in introspection (`IntrospectionInterpreter._add_languages` seems wrong but it can wait after the release). Leaving the f-string is harmless for now.